### PR TITLE
add label function and test

### PR DESCRIPTION
**Pull request by bayern39**
add the labels operations in the repository service and reference test method,depend on #44

### CODE REVIEW

File: src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java [hunk 1]
```diff
@@ -94,6 +95,38 @@
     List<Patch> getPatchesByStatus(Repository repository, PatchStatus status) throws NotFoundException;
 
     /**
+     * Retrieve all labels associated with the provided <code>Patch</code> in <code>Repository</code> object.
+     * @param patch the <code>Patch<code> object whose associated labels should be returned.
+     * @return a list of all matching <code>Label<code> objects, or an empty list if no patches can be found.
+     * @throws NotFoundException if the provided <code>Patch</code> url not consistent with the baseURL.
+     */
+    List<Label> getLabelsfromRepository(Patch patch) throws NotFoundException;
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java]: wrong argument. Should be repository.
getLabelsByRepository(Repository...)

File: src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java [hunk 2]
```diff
@@ -94,6 +95,38 @@
...
+     */
+    void addLabelstoPatch(Patch patch, List<Label> labels) throws NotFoundException;
+
+    /**
+     *Delete a label from the provided <code>Patch</code> object.
+     * @param patch the <code>Patch</code> whose label will be removed.
+     * @param name the <code>Label</code> name will be removed.
+     * @throws NotFoundException if the <code>Label</code> name can not be found in the provided <code>Patch</code>
+     */
+    void removeLabel(Patch patch, String name) throws NotFoundException;
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java]: due to consistency should be 
addLabelToPatch
removeLabelFromPatch

File: src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java [hunk 3]
```diff
@@ -94,6 +95,38 @@
...
+     */
+    List<Label> getLabelsfromRepository(Patch patch) throws NotFoundException;
+
+    /**
+     * Retrieve all labels associated with the provided <code>Patch</code> object.
+     * @param patch the <code>Patch<code> object whose associated labels should be returned.
+     * @return a list of all matching <code>Label<code> objects, or an empty list if no patches can be found.
+     * @throws NotFoundException if the provided <code>Patch</code> url not consistent with the baseURL.
+     */
+    List<Label> getLabelsfromPatch(Patch patch) throws NotFoundException;
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java]: signature: getLabelsFromPatch

File: src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java [hunk 4]
```diff
@@ -94,6 +95,38 @@
...
+
+    /**
+     * Set the labels for the provided <code>Patch</code> object.
+     * @param patch the <code>Patch</code> object whose will be set.
+     * @param labels the <code>Label</code> apply to the <code>Patch</code>
+     * @throws NotFoundException if the <code>Label</code> can not be found in the provided <code>Patch</code>
+     */
+    void addLabelstoPatch(Patch patch, List<Label> labels) throws NotFoundException;
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/spi/RepositoryService.java]: this should be setLabelsToPatch

File: src/main/java/org/jboss/set/aphrodite/repository/services/github/GitHubRepositoryService.java [hunk 5]
```diff
@@ -215,16 +219,92 @@ private Label getLabel(RepositoryId repositoryId, String labelName) throws NotFo
...
+        } catch (IOException e) {
+            Utils.logException(LOG, e);
+        }
+
+        return WRAPPER.pullRequestLabeltoPatchLabel(labels, patchId);
+    }
+
+    @Override
+    public void addLabelstoPatch(Patch patch, List<Label> labels) throws NotFoundException {
+        for (Label label : labels) {
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/repository/services/github/GitHubRepositoryService.java]: this does not fit the description of the service. set labels has/should have the next postcondition:
labels : List<Label>
setLabelsToPatch(labels)
getLabelsFromPatch().equals(labels)

File: src/main/java/org/jboss/set/aphrodite/domain/Label.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.set.aphrodite.domain;
+
+public class Label {
+
+    private String patchId;
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/domain/Label.java]: not needed here. Label is an independent entity from patch.

**Comment by elguardian** [comment]
I added a few comments.

**Comment by bayern39** [comment]
@elguardian i has commited the latest version

File: src/main/java/org/jboss/set/aphrodite/repository/services/github/GitHubRepositoryService.java [hunk 7]
```diff
@@ -215,16 +220,92 @@ private Label getLabel(RepositoryId repositoryId, String labelName) throws NotFo
...
+            labels = issue.getLabels();
+        } catch (IOException e) {
+            Utils.logException(LOG, e);
+        }
+
+        return WRAPPER.pullRequestLabeltoPatchLabel(labels);
+    }
+
+    @Override
+    public void addLabelsToPatch(Patch patch, List<Label> labels) throws NotFoundException, AphroditeException {
```
Comment (elguardian) [inline src/main/java/org/jboss/set/aphrodite/repository/services/github/GitHubRepositoryService.java]: we have two options here. either you keep this call and you create the setLabelsToPatch() or you modify this signature and change this to fit the set call documentation in the issue https://github.com/jboss-set/aphrodite/issues/44 and you set the proper logic in this call.

**Comment by bayern39** [comment]
@elguardian fixed

**Pull request merged**